### DISCUSSION TITLE
Does Corinna include enhancements to the perl debugger?

### DISCUSSION THREAD

**Discussion by BramVan-Oosterhout** [Ovid/Cor#83] [Q&A]
I do not see any discussion of the debug implementation. Is there any support for the debugger planned specific to Corinna? The perl debugger is a great tool to get to know the code of a new (to me) open source project. My experience joining a Moo based perl project failed, because the debugger insisted to go down all the Moo rat holes, instead of following the code as written. I did modify the debugger to remove the biggest diversions, but eventually gave up.

**Comment by Ovid** [comment]
For the issue with Moo, [you can check out my .perldb file](https://gist.github.com/Ovid/919234335d7fc27fca3ec63e6f3782ce). Amongst other things, it lets you skip over modules you don't care about (such as stepping through Moo).

For Corinna, it should mostly work just fine with the debugger because you won't have a deep Perl stack to step through. The only caveat (that I can think of) being that it's not yet clear what will happen when one types `x $corinna_object`. That's probably going to be resolved when the other efforts at serializing Corinna objects are underway.

**Reply by BramVan-Oosterhout** [reply]
Thanks for the reference to the perldb file. I will use that if I ever venture close to another Moo project.

You answer re Corinna is encouraging. i will keep following your progress.
